src/main/scala/diplomaticobjectmodel/logicaltree/LogicalTreeNode.scala

```diff
@@ -11,27 +12,47 @@ trait LogicalTreeNode {
   def getOMComponents(resourceBindingsMap: ResourceBindingsMap, children: Seq[OMComponent] = Nil): Seq[OMComponent]
 }
 
+class RootLogicalTreeNode extends LogicalTreeNode {
```

Comment:
Make one concrete generic LTN and replace this LTN and the ImpLTN with the single concrete LTN.